packages/amplify-graphql-conversation-transformer/src/transformer-steps/conversation-field-handler.ts

```diff
@@ -112,6 +160,26 @@ export class ConversationFieldHandler {
   private validate(config: ConversationDirectiveConfiguration): void {
     this.validateReturnType(config);
     this.validateInferenceConfig(config);
+    this.validateHandler(config);
+  }
+
+  /**
+   * Validates the handler configuration for the conversation directive.
+   * @param {ConversationDirectiveConfiguration} config - The conversation directive configuration to validate.
+   * @throws {InvalidDirectiveError} If the configuration is invalid.
+   */
+  private validateHandler(config: ConversationDirectiveConfiguration): void {
+    if (config.handler && config.functionName) {
```

Comment:
we could remove `functionName` at this point.

Reply:
Yea, I'll follow up with that. This PR is exclusively non-functional changes and considering its size, I'd like to keep it that way.